docs/authentication.md

```diff
@@ -0,0 +1,43 @@
+# Authentication
+
+## Devise
+
+[Devise](https://github.com/heartcombo/devise) is a flexible authentication solution that provides a full-featured authentication system that provides a complete MVC solution based on Rails engines. Some of the modules it includes:
+
+**Database Authenticatable**, **Registerable**, **Confirmable**, **Timeoutable**, **Omniauthable**..etc
+
+Here is an example of how Devise is used in our User model:
+
+```
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable, :confirmable, :trackable, :omniauthable
+end
+```
+
+Devise is customizable, allowing you to add or remove modules that best suit your application. Devise provides many helper methods like `user_signed_in?` that checks if a user is signed in or `current_user` that returns the **User** record of the signed in user.
```

Comment:
we don't need to explain about devise here. we just need to link to the official docs.